plugins/callsites/callsites_main.ml

```diff
@@ -54,14 +54,29 @@ let target intent sub blk call =
     | _ -> None
   else Some blk
 
+(* Note, that we intentionally invert compare for Both and Out
+   arguments, since they will be inserted in reverted order later. *)
+let enum_args t =
+  let compare x y = match Arg.intent x, Arg.intent y with
```

Comment:
looks like this order coincides with the `intent_order`, so it is easier and better just to use it.